gw_eccentricity/eccDefinition.py

```diff
@@ -1237,6 +1279,17 @@ def measure_ecc(self, tref_in=None, fref_in=None):
             apocenters = self.find_extrema("apocenters")
         original_apocenters = apocenters.copy()
         self.check_num_extrema(apocenters, "apocenters")
+
+        # If the eccentricity is too small for a method to find the extrema and
+        # set_failures_to_zero is set to true, then we set the eccentricity and
+        # mean anomaly to zero and return it.
+        # In this case, the rest of the code in this function is not executed and
+        # therefore, many variables which are used in diagnostic tests are never
+        # computed thus making diagnostics irrelevant.
+        if self.probably_quasicircular and self.set_failures_to_zero:
```

Comment:
If this is the reason you have probably_quasicircular in self, it is much clearer to have probably_quasicircular returned by self.check_num_extrema. Otherwise, it's very unclear where what gets set.

Reply:
Okay, done.